realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -1288,15 +1292,1274 @@ public void add(@Nullable E e) {
         }
     }
 
-    /*
-     * MEMO: This method supports {@code long}, {@code  int}, {@code short}, {@code byte},
-     * {@code double}, {@code float} and {@code boolean} as {@code T} in addition to classes
-     * mentioned in the class comment of {@link RealmValueList}.
-     */
-    @Nonnull
     @Override
-    public <T> T[] toArray(@Nonnull T[] a) {
-        // TODO implement this
+    public <T> T[] toArray(T[] a) {
+        if (!isManaged()) {
+            return super.toArray(a);
+        }
+
+        final Class<?> componentType = a.getClass().getComponentType();
+        final int size = size();
+        if (componentType == String.class) {
+            if (a.length < size) {
+                //noinspection unchecked
+                a = (T[]) Array.newInstance(componentType, size);
+            }
+            for (int i = 0; i < size; i++) {
+                //noinspection unchecked
+                a[i] = (T) osListOperator.get(i);
+            }
+            return a;
+        }
+
+        if (componentType == Long.class) {
+            if (a.length < size) {
+                //noinspection unchecked
+                a = (T[]) Array.newInstance(componentType, size);
+            }
+
+            for (int i = 0; i < size; i++) {
+                final E e = osListOperator.get(i);
+                if (e == null || e instanceof Long) {
+                    //noinspection unchecked
+                    a[i] = (T) e;
+                } else {
+                    if (!(e instanceof Number)) {
```

Comment:
Not sure I understand this else block. When you can enter that? Same for the rest

Reply:
the `get(i)` here should never return a non-number value i think?

Reply:
@cmelchior When a user defines the field like `public RealmList<Integer> fooList;`, following call can enter `else` block.

```
Long[] integerList = realmList.toArray(new Long[0]);
```

Should we throw `ArrayStoreException` in this case?

Reply:
removed special implementation of `toArray(T[])`